impl/src/main/java/org/ehcache/UserManagedCacheBuilder.java

```diff
@@ -268,7 +296,46 @@ public void close() throws Exception {
     }
 
   }
-  
+
+  private void validateListenerConfig() {
+    if (!eventListenerConfigurations.isEmpty() && eventDispatcher instanceof DisabledCacheEventNotificationService) {
+      if (orderedExecutor == null && unOrderedExecutor == null) {
+        throw new IllegalArgumentException("Listeners will not work unless Executors or EventDispatcher is configured.");
+      }
+    }
+  }
+
+  private void registerListeners(Cache<K, V> cache, Store<K, V> store, ServiceProvider serviceLocator, List<LifeCycled> lifeCycledList) {
+    if (!eventListenerConfigurations.isEmpty()) {
+      final CacheEventListenerProvider listenerProvider;
+      CacheEventListenerProvider provider;
+      if ((provider = serviceLocator.getService(CacheEventListenerProvider.class)) != null) {
+        listenerProvider = provider;
+      } else {
+        listenerProvider = new DefaultCacheEventListenerProvider(new DefaultCacheEventListenerProviderConfiguration());
```

Comment:
Please get rid of the configuration class. It is useless. The `ClassInstanceProvider` does not need it, it accepts a null as the first parameter in its constructor.